### DISCUSSION TITLE
"Invalid response from authorization hook"

### DISCUSSION THREAD

**Discussion by beebase** [hasura/graphql-engine#7266] [General]
I'm getting this webhook error when I'm trying to connect from a urql client to hasura server which doesn't give me much information to go on. Any idea what might be causing this error?
(when using apollo client I get the same error)

error log hasura server
```
{
"type":"http-log",
"timestamp":"2021-07-20T15:49:07.393+0000",
"level":"error",
"detail":{"operation":{"error":{"path":"$",
"error":"Invalid response from authorization hook",
"code":"unexpected"
}
```

The webhook is working fine when I test with Postman or ThunderClient and is returning exactly what I'd expect
```
{
  "X-Hasura-Role": "admin",
  "X-Hasura-User-Id": "abcdef..."
}
```

URQL CLIENT CONFIG (urql.js)
```
import { SubscriptionClient } from "subscriptions-transport-ws";
import {
  createClient,
  dedupExchange,
  fetchExchange,
  subscriptionExchange,
} from "@urql/svelte";
import { GQL_WS_HOST, GQL_HTTP_HOST } from "@/js/constants"


const token = "abc"
const subscriptionClient = new SubscriptionClient(GQL_WS_HOST, {
  reconnect: true,
  lazy: true,
  connectionParams: {
    headers: {
      authorization: token ? `Bearer ${token}` : ""
    }
  },
});

export const urqlClient = createClient({
  exchanges: [
    // devtoolsExchange,
    dedupExchange,
    // CacheExchange,
    // retryExchange(retryOptions),
    fetchExchange,
    subscriptionExchange({
      forwardSubscription(operation) {
        return subscriptionClient.request(operation);
      },
    }),
  ],
  url: GQL_HTTP_HOST,
  fetchOptions: () => {
    return {
      headers: { authorization: token ? `Bearer ${token}` : "" },
    };
  },
});
```

CLIENT CALL
```
import { urqlClient } from "@/js/urql"
export const querySessionUser =  (_IdSessionUser) => {
  urqlClient.query(sQueryUser, { id: _idSessionUser })
    .toPromise()
    .then(result => console.log(f, 'result', result))
// result is a 200 response with following error
originalError:
extensions: {path: "$", code: "unexpected"}
message: "Invalid response from authorization hook"
```

The webhook is running in a node server on the same (host) server location (linux) as the hasura server (which runs in docker)

The node server uses npm package cors with default setting 
app.use(cors())

Maybe the AUTH_HOOK is not correct? I believe you should use 172.17.0.1 in order to communicate from docker to host machine?
($ ip addr show docker0 will give me 172.17.0.1)

HASURA SERVER SETTINGS
```
...
HASURA_GRAPHQL_AUTH_HOOK: http://172.17.0.1:8888/server/auth/validate
HASURA_GRAPHQL_ADMIN_SECRET: "*****"
....
```

**Comment by praveenweb** [comment]
Have you checked the network request that is being sent when using urql or apollo client? The invalid response could be due to the Authorization token being an empty string.

**Reply by karthickselvendran** [reply]
I am facing the same issue, In network tab Authorization is passing but in webhook request headers is not coming.

Please help to resolve this issue.
